build_runner/test/common/in_memory_reader.dart

```diff
@@ -7,22 +7,16 @@ import 'dart:mirrors';
 import 'package:build/build.dart';
 import 'package:build_runner/build_runner.dart';
 import 'package:build_test/build_test.dart';
+import 'package:glob/glob.dart';
 
 class InMemoryRunnerAssetReader extends InMemoryAssetReader
     implements RunnerAssetReader {
   InMemoryRunnerAssetReader([Map<AssetId, DatedValue> sourceAssets])
       : super(sourceAssets);
 
   @override
-  Stream<AssetId> listAssetIds(Iterable<InputSet> inputSets) async* {
-    for (var id in assets.keys) {
-      var matches = inputSets.any((inputSet) {
-        if (inputSet.package != id.package) return false;
-        return inputSet.globs.any((glob) => glob.matches(id.path));
-      });
-      if (matches) yield id;
-    }
-  }
+  Iterable<AssetId> findAssets(Glob glob, {String packageName}) => assets.keys
```

Comment:
`assert(packageName != null)`? In general this exposes a bit of a weirdness in the api I think. A `BuildStep` definitely has a default package, but that isn't necessarily the case for an arbitrary `AssetReader`. Maybe it doesn't matter though, and we do want them to share the same api.